extensions/stackdriver/log/exporter.cc

```diff
@@ -70,9 +71,23 @@ ExporterImpl::ExporterImpl(RootContext* root_context,
   if (logging_service_endpoint.empty()) {
     grpc_service.mutable_google_grpc()->set_target_uri(
         kGoogleStackdriverLoggingAddress);
-    grpc_service.mutable_google_grpc()
-        ->add_call_credentials()
-        ->mutable_google_compute_engine();
+    if (sts_port.empty()) {
```

Comment:
Nit: just extract this out as “configure_auth” and use it in 2 places.

Reply:
The PR has no tests, can we add something here?

Reply:
Extracted the logic to common util. I am not sure how can I test the stub effectively.

Reply:
We only turn on STS server when port is > 0, can we verify it here as well?